core/src/message/incoming.rs

```diff
@@ -74,12 +71,17 @@ impl IncomingMessage {
             self.id,
             self.source,
             destination,
-            self.payload,
+            Arc::unwrap_or_clone(self.payload),
```

Comment:
shouldn't we make it just unwrap? try_into_strored -> Option<StoredMessage>

Reply:
I think we could get `Option`, but we would have to handle the error situation somehow, here we get what we want anyway even if the `payload` has many hard refs

Reply:
can we at least add here log::warn so we could check if we miss refs somewhere ?